### PR TITLE
v1.4.0 + #1631 (remove go1.6/go1,7 support)

### PR DESCRIPTION
**Pull request by dmarkham**
This is a starting point for the `v1.4.0` release issue #1732.
This also merges in #1631 after fixing the conflicts.

### CODE REVIEW

**Comment by dmarkham** [comment]
@appleboy  @thinkerou I heard a rumor you guys might be able to do releases now. Can I do anything to make this PR more helpful to push this over the finish line!

**Comment by appleboy** [comment]
@dmarkham I have permission to create new tag now. I will review the PR asap

File: context.go [hunk 1]
```diff
@@ -439,11 +439,6 @@ func (c *Context) GetPostFormArray(key string) ([]string, bool) {
 	if values := req.PostForm[key]; len(values) > 0 {
 		return values, true
 	}
-	if req.MultipartForm != nil && req.MultipartForm.File != nil {
-		if values := req.MultipartForm.Value[key]; len(values) > 0 {
-			return values, true
-		}
-	}
```
Comment (appleboy) [inline context.go]: Why remove this condition?

File: context.go [hunk 2]
```diff
@@ -462,13 +457,7 @@ func (c *Context) GetPostFormMap(key string) (map[string]string, bool) {
 			debugPrint("error on parse multipart form map: %v", err)
 		}
 	}
-	dicts, exist := c.get(req.PostForm, key)
-
-	if !exist && req.MultipartForm != nil && req.MultipartForm.File != nil {
-		dicts, exist = c.get(req.MultipartForm.Value, key)
-	}
```
Comment (appleboy) [inline context.go]: same as above.

Comment (thinkerou) [reply context.go]: Add this code for support go1.6, please commit message: https://github.com/gin-gonic/gin/pull/1383/commits/51386de165ae50a61c06d80f06ec703810a2d62d

File: context.go (hunk 1 above)
Comment (thinkerou) [reply context.go]: Add this code for support go1.6, please commit message: https://github.com/gin-gonic/gin/pull/1383/commits/51386de165ae50a61c06d80f06ec703810a2d62d

**Comment by dmarkham** [comment]
:fireworks:

**Pull request merged**